### PR TITLE
Better way to run parallel test processes in different directories

### PR DESCRIPTION
**Pull request by BrettDong**
#### Summary
None

#### Purpose of change
My solution to avoid concurrent filesystem access in #47134 is not ideal. I drilled a hole in the `WORLD` class to make test processes run in unique worlds named by its PID, but @jbytheway suggested a more elegant solution: pass `--user-mod=<dir>` to `cata_test`. I wanted to change it but was distracted by other things.

#### Describe the solution
Repay the technical debt. Adopt @jbytheway's suggestion.

#### Testing
Wait for tests to complete on this pull request.

### CODE REVIEW

**Comment by actual-nh** [comment]
If one is using the parallel mechanism to run the equivalent of a "matrix" - more than one changing substitution - what's the recommended way to do this with --user-dir? --user-dir=something depending on {1} and {2}, I would guess - which would need to remove any filesystem-sensitive characters from the {1} and {2}? See #47791 for one particular case involved.

**Comment by BrettDong** [comment]
What comes to my mind is `$(echo -n {1} | base64`

**Comment by jbytheway** [comment]
If you wanted to make them look as similar as possible to the input but avoid a ew problem characters you could convert problematic characters with something like `sed`, but if you don't care to make it human-readable then I think the `base64` solution is reasonable.

File: build-scripts/build.sh [hunk 1]
```diff
@@ -165,8 +165,8 @@ then
         make -j$num_jobs
         cd ..
         # Run regular tests
-        [ -f "${bin_path}cata_test" ] && parallel --verbose --linebuffer "run_test $(printf %q "${bin_path}")'/cata_test' {} '('{}')=> '" ::: "crafting_skill_gain" "[slow] ~crafting_skill_gain" "~[slow] ~[.]"
-        [ -f "${bin_path}cata_test-tiles" ] && parallel --verbose --linebuffer "run_test $(printf %q "${bin_path}")'/cata_test-tiles' {} '('{}')=> '" ::: "crafting_skill_gain" "[slow] ~crafting_skill_gain" "~[slow] ~[.]"
+        [ -f "${bin_path}cata_test" ] && parallel --verbose --linebuffer --link "run_test $(printf %q "${bin_path}")'/cata_test' {2} {1} '('{1}')=> '" ::: "crafting_skill_gain" "[slow] ~crafting_skill_gain" "~[slow] ~[.]" ::: "--user-dir=process1" "--user-dir=process2" "--user-dir=process3"
```
Comment (Qrox) [inline build-scripts/build.sh:168]: You need to change the `run_test` function to accept the new argument. (Or you can instead append the command line option to the first argument, I suppose.)

**Pull request closed without merging**